internal/component/peerservice/peerservice.go

```diff
@@ -111,7 +111,9 @@ func buildResource(etcd *druidv1alpha1.Etcd, svc *corev1.Service) {
 	svc.Spec.Type = corev1.ServiceTypeClusterIP
 	svc.Spec.ClusterIP = corev1.ClusterIPNone
 	svc.Spec.SessionAffinity = corev1.ServiceAffinityNone
-	svc.Spec.Selector = druidv1alpha1.GetDefaultLabels(etcd.ObjectMeta)
+	// Peer service should only target StatefulSet pods. Default labels are going to be present on anything that is managed by etcd-druid and started for an etcd cluster.
+	// Therefore, only using this as a label selector can cause issues as we have already seen in https://github.com/gardener/etcd-druid/issues/914
```

Comment:
```suggestion
	// Therefore, only using defaultLabels as a label selector can cause issues as we have already seen in https://github.com/gardener/etcd-druid/issues/914
```

Reply:
fixed in 26c5d28cab652dc5c7daac6240bf7fbe116881ae